phpBB/phpbb/storage/controller/controller.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\storage\controller;
+
+use phpbb\cache\service;
+use phpbb\db\driver\driver_interface;
+use phpbb\exception\http_exception;
+use phpbb\storage\storage;
+use Symfony\Component\HttpFoundation\Request as symfony_request;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+/**
+ * Generic controller for storage
+ */
+class controller
+{
+	/** @var service */
+	protected $cache;
+
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var storage */
+	protected $storage;
+
+	/** @var StreamedResponse */
+	protected $response;
+
+	/** @var symfony_request */
+	protected $symfony_request;
+
+	/**
+	 * Constructor
+	 *
+	 * @param service				$cache
```

Comment:
Should be the cache driver interface and not the service

Reply:
Are you sure?

Reply:
I have look at this again, and is service, at least in this case, the class don't extends from driver_interface.
And also i don't know very well when use cache.driver service and when use cache service

Reply:
@cache is the class \phpbb\cache\service that don't implement any interface